zerver/views/realm_playgrounds.py

```diff
@@ -0,0 +1,44 @@
+import re
+
+from django.core.exceptions import ValidationError
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import ugettext as _
+
+from zerver.decorator import require_realm_admin
+from zerver.lib.actions import do_add_realm_playground
+from zerver.lib.request import REQ, JsonableError, has_request_variables
+from zerver.lib.response import json_error, json_success
+from zerver.lib.validator import check_string, check_url
+from zerver.models import UserProfile
+
+
+def check_pygments_language(var_name: str, val: object) -> str:
+    s = check_string(var_name, val)
+    valid_pygments_language = re.compile(r"^[ a-zA-Z0-9_+-./#]*$")
+    matched_results = valid_pygments_language.match(s)
+    if not matched_results:
+        raise JsonableError(_("Invalid characters in pygments language"))
+    return s
+
+
+@require_realm_admin
+@has_request_variables
+def add_realm_playground(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    name: str = REQ(),
+    url_prefix: str = REQ(validator=check_url),
+    pygments_language: str = REQ(validator=check_pygments_language),
+) -> HttpResponse:
+    playground_dict = {
+        "name": name.strip(),
+        "pygments_language": pygments_language.strip(),
+        "url_prefix": url_prefix.strip(),
+    }
```

Comment:
Personally I'd have just made these either positional parameters or required kwargs to `do_add_realm_playground` -- that way the type-checker does more useful validation for you.